File: scenarios/storage-java-crud-blobs/golden/src/main/java/com/example/BlobCrud.java
package com.example;

import com.azure.identity.DefaultAzureCredentialBuilder;
import com.azure.storage.blob.BlobClient;
import com.azure.storage.blob.BlobContainerClient;
import com.azure.storage.blob.BlobServiceClient;
import com.azure.storage.blob.BlobServiceClientBuilder;
import com.azure.storage.blob.models.BlobItem;
import com.azure.storage.blob.models.BlobStorageException;

public final class BlobCrud {
    private static final String CONTAINER_NAME = "my-container";
    private static final String BLOB_NAME = "uploads/data.txt";

    private BlobCrud() {
    }

    public static void main(String[] args) {
        String accountUrl = requireEnvironment("AZURE_STORAGE_ACCOUNT_URL");
        BlobServiceClient serviceClient = new BlobServiceClientBuilder()
                .endpoint(accountUrl)
                .credential(new DefaultAzureCredentialBuilder().build())
                .buildClient();
        BlobContainerClient containerClient =
                serviceClient.getBlobContainerClient(CONTAINER_NAME);
        BlobClient blobClient = containerClient.getBlobClient(BLOB_NAME);

        try {
            if (!containerClient.exists()) {
                containerClient.create();
            }

            blobClient.uploadFromFile("data.txt", true);

            for (BlobItem blob : containerClient.listBlobs()) {
                System.out.printf(
                        "Blob: %s, size: %d bytes%n",
                        blob.getName(),
                        blob.getProperties().getContentLength());
            }

            blobClient.downloadToFile("data-downloaded.txt", true);
            blobClient.delete();
            containerClient.delete();
        } catch (BlobStorageException exception) {
            System.err.printf(
                    "Blob Storage request failed with status %d: %s%n",
                    exception.getStatusCode(),
                    exception.getMessage());
            throw exception;
        }
    }

    private static String requireEnvironment(String name) {
        String value = System.getenv(name);
        if (value == null || value.isBlank()) {
            throw new IllegalStateException(
                    "Set " + name + " before running.");
        }
        return value;
    }
}

File: scenarios/storage-python-crud-blobs/golden/blob_crud.py
from __future__ import annotations

import os
import sys

from azure.core.exceptions import HttpResponseError, ResourceExistsError
from azure.identity import DefaultAzureCredential
from azure.storage.blob import BlobServiceClient


CONTAINER_NAME = "my-container"
BLOB_NAME = "reports/report.csv"
SOURCE_PATH = "report.csv"
DOWNLOAD_PATH = "report-downloaded.csv"


def run() -> None:
    account_url = os.environ["AZURE_STORAGE_ACCOUNT_URL"]

    with DefaultAzureCredential() as credential:
        with BlobServiceClient(
            account_url=account_url,
            credential=credential,
        ) as service_client:
            container_client = service_client.get_container_client(CONTAINER_NAME)
            try:
                try:
                    container_client.create_container()
                except ResourceExistsError:
                    pass

                blob_client = container_client.get_blob_client(BLOB_NAME)
                with open(SOURCE_PATH, "rb") as source:
                    blob_client.upload_blob(source, overwrite=True)

                for blob in container_client.list_blobs():
                    print(f"{blob.name}: {blob.size}")

                with open(DOWNLOAD_PATH, "wb") as destination:
                    blob_client.download_blob().readinto(destination)

                blob_client.delete_blob()
                container_client.delete_container()
            except HttpResponseError as error:
                print(f"Blob Storage request failed: {error}", file=sys.stderr)
                raise


if __name__ == "__main__":
    run()
